### PR TITLE
Fix kill count

### PR DESCRIPTION
**Pull request by AminArria**
## Motivation

Kill count was broken

## Summary of changes

Fix kill count by changing `grant_power_up_to_killer/3` params (see [comment](https://github.com/lambdaclass/mirra_backend/pull/953/files#r1757389974) for explanation)

## How to test it?

Play the game in unity and you should see the kill count working

## Checklist
- [x] Tested the changes locally.
- [x] Reviewed the changes on GitHub, line by line.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.

### CODE REVIEW

File: apps/arena/lib/arena/game_updater.ex [hunk 1]
```diff
@@ -532,27 +532,27 @@ defmodule Arena.GameUpdater do
...
       ) do
     entry = %{killer_id: killer_id, victim_id: victim_id}
-    victim = Map.get(game_state.players, victim_id)
-    killer = Map.get(game_state.players, killer_id)
 
     game_state =
       game_state
       |> update_in([:killfeed], fn killfeed -> [entry | killfeed] end)
       |> maybe_add_kill_to_player(killer_id)
-      |> grant_power_up_to_killer(game_config, killer, victim)
```
Comment (AminArria) [inline apps/arena/lib/arena/game_updater.ex:542]: The problem here is that `maybe_add_kill_to_player/2` was updating the `killer` inside `game_state`, but then `grant_power_up_to_killer/3` was using the old `killer` binded outside the pipeping, in essence, overriding any changes that happened to `killer` after that binding

Comment (tvillegas98) [reply apps/arena/lib/arena/game_updater.ex:542]: Nice catch!

Comment (tvillegas98) [reply apps/arena/lib/arena/game_updater.ex:542]: Immutable data played all of us while reviewing 😆

**Review by tvillegas98 — APPROVED** [review]

**Review by agustinesco — APPROVED** [review]
LGTM!

**Pull request merged**